static/js/markdown.js

```diff
@@ -108,7 +108,7 @@ export function apply_markdown(message) {
             let full_name;
             let user_id;
 
-            const id_regex = /(.+)\|(\d+)$/g; // For @**user|id** syntax
+            const id_regex = /(.+)?\|(\d+)$/g; // For @**user|id** and @**|id** syntax
```

Comment:
This regex is missing a `^` anchor. I’m guessing the reason it wasn’t there is it was copied from the Python side and Python’s `re.match` (unlike `re.search`) implicitly anchors to the start.

The `g` flag needlessly marks the regex as stateful and should be removed.

Reply:
Yeah I added `^` anchor and removed `g` flag. I did made these changes because, I mostly avoided changing already existing patterns and also added new patterns in somewhat matching fashion.